extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -767,6 +770,25 @@ export class AmpStoryPage extends AMP.BaseElement {
     return null;
   }
 
+  /**
+   * Finds any elements in the page that has a goToPage action.
+   * @return {Array<string>} The IDs of the potential next pages in the story
+   * or [] if there aren't any.
+   */
+  getNextPagesByAction() {
+    const actionElements =
+      Array.from(
+          this.element.querySelectorAll('[on]'),
+          el => el.getAttribute('on'));
+
+    return actionElements.reduce((res, el) => {
+      if (el.includes('goToPage')) {
+        // The pageId is in between the equals sign & closing parenthesis.
+        res.push(el.slice(el.search('\=(.*)') + 1,-1));
```

Comment:
Will this still work if the element has two actions with parameters? e.g. 
```
on="tap:foo.barAction(baz=1), story.goToPage(id=page-id)"
```

Reply:
Thank you for pointing this out! You're right, this won't work for that case.  It also won't work for handling multiple events either : `on="tap:story.goToPage(id=pageID);submit-error:lightbox2"`.

Made some changes to account for this.

Reply:
Maybe we should also add unit tests for those cases?

Reply:
Good point. I just added some tests to test out the method I wrote.